packages/x-charts/src/PieChart/PiePlot.tsx

```diff
@@ -51,14 +55,14 @@ export interface PiePlotProps {
   slotProps?: PiePlotSlotComponentProps;
   /**
    * Callback fired when a pie item is clicked.
-   * @param {string | number} pieItemIdentifier The pie item identifier.
-   * @param {DefaultizedPieValueType} item The pie item.
    * @param {React.MouseEvent<SVGPathElement, MouseEvent>} event The event source of the callback.
+   * @param {PieItemIdentifier} pieItemIdentifier The pie item identifier.
+   * @param {DefaultizedPieValueType} item The pie item.
    */
   onClick?: (
-    pieItemIdentifier: string | number | undefined,
-    item: DefaultizedPieValueType,
     event: React.MouseEvent<SVGPathElement, MouseEvent>,
+    pieItemIdentifier: PieItemIdentifier,
+    item: DefaultizedPieValueType,
```

Comment:
I modified the order. 

Instead of using the pie slice id which is not mandatory to simplify the creation of basics pie charts (maybe it should be), I used the `PieItemIdentifier` which contains:
- The type of the series ('pie')
- The series id
- The index of the pie slice

At least having the series' id is necessary if you plot a pie chart with multiple pie like for example the [two level pie example](https://mui.com/x/react-charts/pie-demo/#twolevelpiechart)